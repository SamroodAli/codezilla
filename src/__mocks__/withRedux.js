import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import PropTypes from 'prop-types';
import data from '../../public/data/courses.json';

const initialState = {
  videos: [data],
  filters: {
    React: true,
    Angular: true,
    Vue: true,
    'Functional Programming': true,
    'Object Oriented Programming': true,
    Microservices: true,
    'JAM Stack': true,
  },
  theme: 'light',
  currentVideo: null,
};

const mockStore = configureStore([reduxThunk]);
export const store = mockStore(initialState);

const WithRedux = ({ children }) => (
  <Provider store={store}><StaticRouter>{children}</StaticRouter></Provider>
);

WithRedux.propTypes = {
  children: PropTypes.node.isRequired,
};

export default WithRedux;
