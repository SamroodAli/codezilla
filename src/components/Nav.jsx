import * as React from 'react';
import {
  HeaderNavigation,
  ALIGN,
  StyledNavigationList,
  StyledNavigationItem,
} from 'baseui/header-navigation';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Nav = ({ button }) => (
  <HeaderNavigation>
    <StyledNavigationList $align={ALIGN.left}>
      <StyledNavigationItem>
        <Link to="/">
          COdZilla
        </Link>
      </StyledNavigationItem>
    </StyledNavigationList>
    <StyledNavigationList $align={ALIGN.center} />
    <StyledNavigationList $align={ALIGN.right}>
      <StyledNavigationItem>
        <Link to="/courses">
          Courses
        </Link>
      </StyledNavigationItem>
    </StyledNavigationList>
    <StyledNavigationList $align={ALIGN.right}>
      <StyledNavigationItem>
        {button}
      </StyledNavigationItem>
    </StyledNavigationList>
  </HeaderNavigation>
);

Nav.propTypes = {
  button: PropTypes.node.isRequired,
};

export default Nav;
