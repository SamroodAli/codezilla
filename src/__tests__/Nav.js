import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import withRedux from 'withRedux';
import Nav from '../components/Nav';

describe('Testing Nav with redux', () => {
  it('Nav is defined', async () => {
    const component = render(withRedux(<Nav />));
    expect(component).toBeDefined();
  });

  it('Nav snapshot testing', () => {
    const tree = renderer.create(withRedux(<Nav />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
