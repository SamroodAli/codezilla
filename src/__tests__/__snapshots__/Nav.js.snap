// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Nav with redux Nav snapshot testing 1`] = `
<nav
  className=""
  data-baseweb="header-navigation"
  role="navigation"
>
  <ul
    className=""
  >
    <li
      className=""
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className=""
          data-baseweb="typo-displaymedium"
        >
          CodeZilla
        </div>
      </a>
    </li>
  </ul>
  <ul
    className=""
  />
  <ul
    className=""
  >
    <li
      className=""
    >
      <a
        href="/courses"
        onClick={[Function]}
      >
        <div
          className=""
          data-baseweb="typo-labellarge"
        >
          Courses
        </div>
      </a>
    </li>
  </ul>
  <ul
    className=""
  >
    <li
      className=""
    />
  </ul>
</nav>
`;
